src/main/java/org/traccar/protocol/TranSyncProtocolDecoder.java

```diff
@@ -0,0 +1,218 @@
+package org.traccar.protocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.Protocol;
+import org.traccar.helper.BitUtil;
+import org.traccar.helper.UnitsConverter;
+import org.traccar.model.CellTower;
+import org.traccar.model.Network;
+import org.traccar.model.Position;
+import org.traccar.session.DeviceSession;
+
+import java.net.SocketAddress;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+public class TranSyncProtocolDecoder extends BaseProtocolDecoder {
+
+    public TranSyncProtocolDecoder(Protocol protocol) {
+        super(protocol);
+    }
+
+    public Date getDatefromIntegerParameters(int year, int month, int day, int hours, int minutes, int seconds) {
+
+        String dataString = ""+year + "-" + month + "-" + day + " " + hours + ":" + minutes + ":" + seconds;
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yy-MM-dd HH:mm:ss");
+        simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+        try {
+            return simpleDateFormat.parse(dataString);
+
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    protected void setDeviceAlert(Position position, int alert) {
+        switch (alert) {
+            case 10:
+                position.set(Position.ALARM_SOS, true);
+                break;
+            case 11:
+                position.set(Position.ALARM_SOS, false);
+                break;
+            case 16:
+                position.set(Position.KEY_EVENT, 16);
+                break;
+            case 3:
+                position.set("distressCutOff", true);
+                break;
+            case 22:
+                position.set("tilt", true);
+                break;
+            case 9:
+                position.set(Position.KEY_EVENT, 9);
+            case 17:
+                position.set(Position.ALARM_OVERSPEED, true);
+                break;
+            case 13:
+                position.set(Position.ALARM_BRAKING, true);
+                break;
+            case 14:
+                position.set(Position.ALARM_ACCELERATION, true);
+                break;
+            case 15:
+                position.set(Position.KEY_EVENT, 15);
```

Comment:
It doesn't make sense to do something like this. Instead you can just save all event codes as a one liner.